src/chainrules.jl

```diff
@@ -150,3 +150,40 @@ function ChainRulesCore.rrule(::typeof(ifftshift), x::AbstractArray, dims)
     end
     return y, ifftshift_pullback
 end
+
+# plans
+function ChainRulesCore.frule((_, _, Δx), ::typeof(*), P::Plan, x::AbstractArray) 
+    y = P * x 
+    Δy = P * Δx
+    return y, Δy
+end
+function ChainRulesCore.rrule(::typeof(*), P::Plan, x::AbstractArray)
+    y = P * x
+    project_x = ChainRulesCore.ProjectTo(x)
+    Pt = P'
+    function mul_plan_pullback(ȳ)
+        x̄ = project_x(Pt * ȳ)
+        return ChainRulesCore.NoTangent(), ChainRulesCore.NoTangent(), x̄
```

Comment:
I wonder if there's a better default than `NoTangent` for `Plan`? The example with `ScaledPlan` below shows that it is not always correct. But I guess we just have to demand that users implement their own `rrule` if they use a differentiable plan.

Reply:
Yup, it's unfortunate. I'm not sure if there's a generic way to handle possibly differentiable plans: it's an issue for the `frule` too. When writing this PR I've often found myself wishing there was a way of differentiating the "innermost" plans from the wrapper plans, but that doesn't seem possible with the current type hierarchy.

Reply:
Yes, would be nice to address that in some future PR (in whatever way turns out to be most reasonable).